### PR TITLE
docs: update quickstart to use cloud memorystore

### LINKED ISSUE #1083: Change docs Quickstart to use CloudMemorystoreInstance
### What problem are you facing?

The current [quickstart](https://crossplane.io/docs/v0.5/quick-start.html) walks through statically provisioning a `CloudSQLInstance`. However, this can fail due to the fact that GCP does not allow reuse of instance name [for a week following deletion](https://cloud.google.com/sql/docs/mysql/delete-instance).

### How could Crossplane help solve your problem?

The quickstart should be updated to use a resource that does not have this same constraint. A simple solution would be to provision a GCP `CloudMemorystoreInstance`.

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Fixes #1083 

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation], [examples], or [release notes].
- [x] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

[documentation]: https://github.com/crossplaneio/crossplane/tree/master/docs
[examples]: https://github.com/crossplaneio/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplaneio/crossplane/tree/master/PendingReleaseNotes.md
[`clusterrole.yaml`]: https://github.com/crossplaneio/crossplane/blob/master/cluster/charts/crossplane/templates/clusterrole.yaml

### CODE REVIEW

File: docs/quick-start.md [hunk 1]
```diff
@@ -5,16 +5,18 @@ weight: 210
 ---
 # Getting Started
 
-This guide will demonstrate using Crossplane to deploy a portable MySQL database
-on the Google Cloud Platform (GCP). It serves as an initial introduction to
-Crossplane, but only displays a small set of its features.
+This guide will demonstrate using Crossplane to deploy a portable Cloud
+Memorystore on the Google Cloud Platform (GCP). It serves as an initial
```
Comment (negz) [inline docs/quick-start.md]: Portable Redis cluster would be more in line with the previous documentation.

File: docs/quick-start.md [hunk 2]
```diff
@@ -80,278 +82,274 @@ Save the above `Provider` as `provider.yaml`, save your Google Application
...
-      Ip Configuration:
-        ipv4Enabled:  true
-      Location Preference:
-        Zone:               us-west2-b
-      Pricing Plan:         PER_USE
-      Replication Type:     SYNCHRONOUS
-      Storage Auto Resize:  true
-      Tier:                 db-n1-standard-1
+    Alternative Location Id:  us-west2-b
+    Authorized Network:       projects/crossplane-playground/global/networks/default
```
Comment (negz) [inline docs/quick-start.md]: We should avoid leaking our playground project's name in our docs and examples. I suggest finding and replacing `crossplane-playground` with some other value.

File: docs/quick-start.md [hunk 3]
```diff
@@ -80,278 +82,274 @@ Save the above `Provider` as `provider.yaml`, save your Google Application
...
-labels]. Resource claims like `MySQLInstance` can match different kinds of
-resource class using label selectors, so you could just as easily use the
-exact same `MySQLInstance` to create an Amazon Relational Database Service (RDS)
-instance by creating an `RDSInstanceClass` labelled as `guide: getting-started`.
-When multiple resource classes match the class selector, a matching class is
-chosen at random. Claims can be matched to classes by either:
+labels]. Resource claims like `RedisCluster` can match different kinds of
+resource class using label selectors, so you could just as easily use the exact
+same `RedisCluster` to create an Amazon Replication Group instance by creating a
+`ReplicationGroupsClass` labelled as `guide: getting-started`. When multiple
```
Comment (negz) [inline docs/quick-start.md]:
```suggestion
`ReplicationGroupClass` labelled as `guide: getting-started`. When multiple
```

**Comment by hasheddan** [comment]
Thanks for the review @negz! All comments addressed :)

**Comment by hasheddan** [comment]
@negz @jbw976 do you think this is worth backporting to 0.5?

**Comment by negz** [comment]
I think the problem this is solving is that if someone attempts to follow the static provisioning section of the CloudSQL quick start guide _multiple times_ they'll encounter an error on the subsequent attempts due to CloudSQL resource names not being reusable for a few days after the old resource is deleted. The error is easily worked around by using a different resource name, but it's not obvious that you need to do that.

Given it's straightforward to backport this commit to the 0.5 docs branch we may as well do so, but it doesn't feel like a priority to me.

**Pull request merged**